first_app/hello.py

```diff
@@ -0,0 +1,175 @@
+import jwt
+import os
+import datetime
+import json
+from sqlalchemy import insert
+from functools import wraps
+from flask import Flask, request, render_template, flash
+from werkzeug.security import check_password_hash, generate_password_hash
+from . import create_app
+from .db import get_db, db_session
+from first_app.models import User, Post
+app = create_app()
+def login_required(f):
+    @wraps(f)
+    def _wrapper(*args,**kwargs):
+        
+        user_id = int(kwargs.get("user_id"))
+        access_token = request.headers.get("Authorization")
+        payload = jwt.decode(access_token, app.config['SECRET_KEY'], algorithms='HS256')
+        token_user_id = payload["user_id"]
+        
+        if not token_user_id or user_id != token_user_id:
+            return {"error":"Користувач не авторизований"}, 403
+        
+        user = User.query.filter(User.id==token_user_id).one()
+        #db = get_db()
+        #user = db.execute(
+        #    "SELECT * FROM user WHERE id = ?",
+        #    (token_user_id,),
+        #).fetchone()
+
+        if not user:
+            return {"error": f"Користувач не існує {user_id}"}, 404
+        if int(token_user_id) != user.id:
+            return{"error": "Користувач створює не свій пост"}, 403
+        
+        return f(*args, **kwargs)
+    return _wrapper
+    
+@app.route("/api/v1/register-user", methods=['GET','POST'])
+def registrate_user_api():
+    data = request.json
```

Comment:
1) Check if user with phone number exists in db.
2) If exists -> raise error with 400 status_code and body {"error": "Phone number already exists"}